subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildSessionScopeServices.java

```diff
@@ -140,6 +140,13 @@ CrossProjectConfigurator createCrossProjectConfigurator(BuildOperationExecutor b
     ProjectCacheDir createCacheLayout(StartParameter startParameter, BuildLayoutFactory buildLayoutFactory, ProgressLoggerFactory progressLoggerFactory) {
         BuildLayout buildLayout = buildLayoutFactory.getLayoutFor(new BuildLayoutConfiguration(startParameter));
         File cacheDir = startParameter.getProjectCacheDir() != null ? startParameter.getProjectCacheDir() : new File(buildLayout.getRootDirectory(), ".gradle");
+        /*
+         * add cacheDir to default exclusions because not doing so causes and MD5 error
```

Comment:
Please don't use inline comments, as they go stale. Use class or method comments instead.

This register/unregister logic can probably live in ProjectCacheDir itself and be explained there.